owncloudApp/src/androidTest/java/com/owncloud/android/settings/security/OCSettingsSecurity.kt

```diff
@@ -0,0 +1,244 @@
+
+    private lateinit var mPrefPasscode: CheckBoxPreference
+    private lateinit var mPrefPattern: CheckBoxPreference
+    private lateinit var mPrefTouches: CheckBoxPreference
+
+    private val context = InstrumentationRegistry.getInstrumentation().targetContext
+
+    private val KEY_CHECK_RESULT = "KEY_CHECK_RESULT"
+    private val KEY_PASSCODE = "KEY_PASSCODE"
+    private val KEY_PATTERN = "KEY_PATTERN"
+    private val KEY_CHECK_PATTERN_RESULT = "KEY_CHECK_PATTERN_RESULT"
+
+    @Before
+    fun setUp() {
+        //Only interested in "More" section, so we can get rid of the other categories. SmoothScroll is not
+        //working fine to reach the bottom of the screen, so this approach was taken to display the section
+        val preferenceScreen = activityRule.activity.getPreferenceScreen() as PreferenceScreen
+        val cameraUploadsCategory =
+            activityRule.activity.findPreference("camera_uploads_category") as PreferenceCategory
+        val moreCategory =
+            activityRule.activity.findPreference("more") as PreferenceCategory
+        preferenceScreen.removePreference(cameraUploadsCategory)
+        preferenceScreen.removePreference(moreCategory)
+
+        mPrefPasscode = activityRule.activity.findPreference("set_pincode") as CheckBoxPreference
+        mPrefPattern = activityRule.activity.findPreference("set_pattern") as CheckBoxPreference
+        mPrefTouches = activityRule.activity.findPreference("touches_with_other_visible_windows")
+                as CheckBoxPreference
+
+        Intents.init()
+    }
+
+    @After
+    fun tearDown(){
+        Intents.release()
+        //clear SharedPreferences
+        PreferenceManager.getDefaultSharedPreferences(context).edit().clear().commit()
+    }
+
+    @Test
+    fun securityView(){
+        onView(withText(R.string.prefs_passcode)).check(matches(isDisplayed()))
+        onView(withText(R.string.prefs_pattern)).check(matches(isDisplayed()))
+        onView(withText(R.string.prefs_fingerprint)).check(matches(isDisplayed()))
+        onView(withText(R.string.prefs_fingerprint_summary)).check(matches(isDisplayed()))
+        onView(withText(R.string.prefs_fingerprint)).check(matches(not(isEnabled())))
+        onView(withText(R.string.prefs_touches_with_other_visible_windows)).check(matches(isDisplayed()))
+        onView(withText(R.string.prefs_touches_with_other_visible_windows_summary)).check(matches(isDisplayed()))
+    }
+
+    @Test
+    fun passcodeOpen(){
+        onView(withText(R.string.prefs_passcode)).perform(click())
+        intended(hasComponent(PassCodeActivity::class.java.name))
+    }
+
+    @Test
+    fun patternOpen(){
+        onView(withText(R.string.prefs_pattern)).perform(click())
+        intended(hasComponent(PatternLockActivity::class.java.name))
+    }
+
+    @Test
+    fun passcodeLockEnabled(){
+        val result = Intent()
+        result.putExtra(KEY_PASSCODE, "1111")
```

Comment:
Save in "1111" in a constant? `PASSCODE_VALUE`

Reply:
Done 👍